nullaway/src/main/java/com/uber/nullaway/handlers/InferredJARModelsHandler.java

```diff
@@ -196,11 +196,15 @@ public boolean onOverrideMayBeNullExpr(
       @Nullable Symbol exprSymbol,
       VisitorState state,
       boolean exprMayBeNull) {
-    if (expr.getKind().equals(Tree.Kind.METHOD_INVOCATION)
-        && exprSymbol instanceof Symbol.MethodSymbol) {
-      return exprMayBeNull || isReturnAnnotatedNullable((Symbol.MethodSymbol) exprSymbol);
+    if (exprMayBeNull) {
+      return true;
     }
-    return exprMayBeNull;
+    if (expr.getKind() == Tree.Kind.METHOD_INVOCATION
+        && exprSymbol instanceof Symbol.MethodSymbol
+        && isReturnAnnotatedNullable((Symbol.MethodSymbol) exprSymbol)) {
+      return true;
+    }
+    return false;
```

Comment:
this is structured like:
```
if reason1_why_expr_may_be_null:
  return true
if reason2_why_expr_may_be_null:
  return true
return false
```

when there is only a single reason, we could of course use the condition in a single return statement if preferred?

the old style is imo more confusing (an has redundant checks):
```
if check1:
  return exprMayBeNull || check2
return exprMayBeNull
```

if any of the checks have important side-effects we should add tests that expose those.

Reply:
agreed this is an improvement here